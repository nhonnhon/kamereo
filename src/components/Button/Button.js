import React from 'react'
import PropTypes from 'prop-types'

import './Button.scss'

const textMapper = {
  white: 'white',
  gray: 'gray',
  black: 'black',
}

const Button = props => {
  const { label, size, color, className, children, onClick, link, textColor, bold } = props

  const classNames = `btn  ${size || ''} ${color || ''} ${className || ''}`

  return (
    <a href={link || 'javascript: void(0)'} className={classNames} onClick={onClick}>
      {children ? (
        children
      ) : (
        <span className={`text ${bold === true ? 'bold' : ''} ${textColor || textMapper[color]} ${size}`}>
          {label}
        </span>
      )}
    </a>
  )
}

Button.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.oneOf(['small', 'big']),
  color: PropTypes.oneOf(['white', 'black', 'gray', 'gray1', 'green', 'transparent']),
  label: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.any,
}

Button.defaultProps = {
  className: '',
  color: '',
}

export default Button
