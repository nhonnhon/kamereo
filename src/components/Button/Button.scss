@import '../../assets/css/base.scss';

.btn {
  display: inline-block;
  outline: none;
  padding: 15px 15px;
  background: transparent;
  text-align: center;
  border: 0;
  cursor: pointer;
  text-decoration: none;
  &:hover {
    opacity: 0.8;
  }
  &.big {
    display: block;
  }
  &.small {
    min-width: 220px;
  }
  &.white {
    background: $white;
  }
  &.gray {
    background: $gray;
  }
  &.gray1 {
    background: $gray1;
  }
  &.black {
    background: $black;
  }
  &.green {
    background: $green;
  }
  &.transparent {
    background: transparent;
  }
  &.disable {
    cursor: default;
  }
  .text {
    text-align: center;
  }
}
