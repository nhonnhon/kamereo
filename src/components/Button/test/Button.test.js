import React from 'react'
import { create } from 'react-test-renderer'
import Button from '../Button'

describe('Button component', () => {
  test('Matches the snapshot', () => {
    const button = create(<Button color="white" size="big" />)
    expect(button.toJSON()).toMatchSnapshot()
  })
})
