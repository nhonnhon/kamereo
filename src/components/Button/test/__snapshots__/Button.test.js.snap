// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<a
  className="btn  big white "
  href="javascript: void(0)"
>
  <span
    className="text  white big"
  />
</a>
`;
